### PR TITLE
Make transpose work with infinite lists

### LINKED ISSUE #732: Transpose (`∩`) needs infinite list support
With itertools, infinite lists don't work at all. [Try it online](https://vyxal.pythonanywhere.com/#WyIiLCIiLCLDnuKInsabw57iiJ474oipIiwiIiwiIl0=)

### PR DESCRIPTION
**Pull request by cgccuser**
Fixes #732. This is probably much more inefficient than before, but it should work for finite lists of finite lists, finite lists of infinite lists, infinite lists of finite lists, and infinite lists of infinite lists.

### CODE REVIEW

**Comment by cgccuser** [comment]
How did black fail lol, I formatted it before committing

File: vyxal/helpers.py [hunk 1]
```diff
@@ -770,14 +770,34 @@ def transfer_capitalisation(source: str, target: str) -> str:
...
+        c = 0
+        while has_ind(matrix, c):
+            if has_ind(matrix[c], r):
+                yield matrix[c][r]
+            elif filler is not None:
+                yield filler
+            c += 1
+
+    r = 0
+    while r < 10:
```
Comment (Lyxal) [inline vyxal/helpers.py:795]: What exactly does this do?

Comment (cgccuser) [reply vyxal/helpers.py:795]: Oh shoot, that's debugging code, it should be `while True:`

**Review by Lyxal — APPROVED** [review]
lgtm

File: tests/test_complex.py [hunk 2]
```diff
@@ -282,3 +282,12 @@ def test_tilde_dyad():
 def test_vectorised_nilad():
     stack = run_vyxal("123 f vkd")
     assert stack[-1][:3] == ["0123456789", "0123456789", "0123456789"]
+
+
+def test_transpose_inf():
+    stack = run_vyxal("Þ∞ ƛÞ∞ +; ∩")
+    assert [row[:3] for row in stack[-1][:3]] == [
+        [2, 3, 4],
```
Comment (cgccuser) [inline tests/test_complex.py:290]: In retrospect, this is not the best test ever.

**Comment by cgccuser** [comment]
Oops

**Review by Seggan — APPROVED** [review]
Why me?

**Comment by cgccuser** [comment]
@Seggan I asked for your review because I saw you were active in chat, so you were likely to see it, and because I needed a review from someone.

**Pull request merged**